com.sap.cloud.lm.sl.cf.web/src/test/java/com/sap/cloud/lm/sl/cf/web/security/CustomTokenServicesTest.java

```diff
@@ -0,0 +1,153 @@
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+
+import com.sap.cloud.lm.sl.cf.client.util.TokenProperties;
+import com.sap.cloud.lm.sl.cf.core.auditlogging.AuditLoggingFacade;
+import com.sap.cloud.lm.sl.cf.core.auditlogging.AuditLoggingProvider;
+import com.sap.cloud.lm.sl.cf.core.security.token.TokenParserChain;
+import com.sap.cloud.lm.sl.cf.core.util.ApplicationConfiguration;
+import com.sap.cloud.lm.sl.cf.core.util.SecurityUtil;
+
+public class CustomTokenServicesTest {
+
+    private static final String DUMMY_TOKEN_STRING = "dummyTokenString";
+    private static final String DEFAULT_SCOPE = "default";
+    private static final String TEST_CLIENT_ID = "testClientId";
+    private static final String USER_NAME = "testUser";
+    private static final String USER_ID = "1";
+
+    @Mock
+    private TokenStore tokenStore;
+    @Mock
+    private TokenParserChain tokenParserChain;
+    @Mock
+    private ApplicationConfiguration configuration;
+    @Mock
+    private AuditLoggingFacade auditLoggingFacade;
+
+    private CustomTokenServices customTokenServices;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        AuditLoggingProvider.setFacade(auditLoggingFacade);
+        customTokenServices = new CustomTokenServices(tokenStore, configuration, tokenParserChain);
+    }
+
+    @Test
+    public void testGetTokenFromCache() {
+        OAuth2AccessToken token = buildValidToken();
+        OAuth2Authentication auth = buildAuthentication(token);
+        prepareTokenParserChain(token);
+        prepareTokenStore(token, auth);
+
+        OAuth2Authentication loadedAuthentication = customTokenServices.loadAuthentication(DUMMY_TOKEN_STRING);
+
+        assertEquals(auth.getAuthorities(), loadedAuthentication.getAuthorities());
+        assertEquals(auth.getCredentials(), loadedAuthentication.getCredentials());
+        assertEquals(auth.getOAuth2Request(), loadedAuthentication.getOAuth2Request());
+    }
+
+    private OAuth2AccessToken buildValidToken() {
+        long expirationTime = LocalDateTime.now()
+                                           .plusMinutes(10)
+                                           .atZone(ZoneId.systemDefault())
+                                           .toEpochSecond();
+        Map<String, Object> additionalTokenProperties = buildAdditionalTokenProperties();
+        return buildToken(DUMMY_TOKEN_STRING, new Date(TimeUnit.SECONDS.toMillis(expirationTime)), additionalTokenProperties);
+    }
+
+    private OAuth2AccessToken buildToken(String tokenString, Date expiration, Map<String, Object> additionalTokenProperties) {
+        DefaultOAuth2AccessToken token = new DefaultOAuth2AccessToken(tokenString);
+        token.setExpiration(expiration);
+        token.setScope(new HashSet<>(Arrays.asList(DEFAULT_SCOPE)));
```

Comment:
If the token's scope will not be modified by underlying code, Collections.singleton(DEFAULT_SCOPE) could be used